llama_stack/apis/inference/inference.py

```
@@ -402,6 +402,34 @@ class ModelStore(Protocol):
     def get_model(self, identifier: str) -> Model: ...
 
 
+class TextTruncation(Enum):
+    """Config for how to truncate text for embedding when text is longer than the model's max sequence length. Start and End semantics depend on whether the language is left-to-right or right-to-left.
+
+    :cvar start: Truncate from the start
+    :cvar end: Truncate from the end
+    """
+
+    start = "start"
+    end = "end"
+
+
+class EmbeddingContext(Enum):
+    """How is the embedding being used? This is only supported by asymmetric embedding models.
+
+    :cvar query: Used for a query for semantic search.
+    :cvar document: Used at indexing time when ingesting documents.
+    """
+
+    query = "query"
+    document = "document"
+
+
+class EmbeddingOptions(BaseModel):
+    dimensions: Optional[int] = None
+    text_truncation: Optional[TextTruncation] = None
```

no image truncation is provided here, images are naturally pre-processed (which means resized / cropped) depending on the settings of each model typically. when a specific need arises to control this pre-processing, we may add an option here. cc @mattf

---

What's the default behavior? As a n00b I was surprised by the token limit, so I would've preferred this to throw an exception if the input is over the token limit, instead of truncating silently.

---

@ehhuang default behavior should be to throw.